src/test/java/uk/gov/pay/api/utils/mocks/ChargeResponseFromConnector.java

```diff
@@ -0,0 +1,267 @@
+    private final SupportedLanguage language; 
+    private final boolean delayedCapture; 
+    private final RefundSummary refundSummary; 
+    private final SettlementSummary settlementSummary; 
+    private final CardDetails cardDetails;
+    private final List<Map<?, ?>> links;
+    private final Optional<Map<String, Object>> metadata;
+
+    public Long getAmount() {
+        return amount;
+    }
+
+    public Long getCorporateCardSurcharge() {
+        return corporateCardSurcharge;
+    }
+
+    public Long getTotalAmount() {
+        return totalAmount;
+    }
+
+    public PaymentState getState() {
+        return state;
+    }
+
+    public String getChargeId() {
+        return chargeId;
+    }
+
+    public String getReturnUrl() {
+        return returnUrl;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public String getReference() {
+        return reference;
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public String getPaymentProvider() {
+        return paymentProvider;
+    }
+
+    public String getGatewayTransactionId() {
+        return gatewayTransactionId;
+    }
+
+    public String getCreatedDate() {
+        return createdDate;
+    }
+
+    public SupportedLanguage getLanguage() {
+        return language;
+    }
+
+    public boolean isDelayedCapture() {
+        return delayedCapture;
+    }
+
+    public RefundSummary getRefundSummary() {
+        return refundSummary;
+    }
+
+    public SettlementSummary getSettlementSummary() {
+        return settlementSummary;
+    }
+
+    public CardDetails getCardDetails() {
+        return cardDetails;
+    }
+
+    public List<Map<?, ?>> getLinks() {
+        return links;
+    }
+
+    public Optional<Map<String, Object>> getMetadata() {
+        return metadata;
+    }
+
+    private ChargeResponseFromConnector(ChargeResponseFromConnectorBuilder builder) {
+        this.amount = builder.amount;
+        this.chargeId = builder.chargeId;
+        this.state = builder.state;
+        this.returnUrl = builder.returnUrl;
+        this.description = builder.description;
+        this.reference = builder.reference;
+        this.email = builder.email;
+        this.paymentProvider = builder.paymentProvider;
+        this.gatewayTransactionId = builder.gatewayTransactionId;
+        this.createdDate = builder.createdDate;
+        this.language = builder.language;
+        this.delayedCapture = builder.delayedCapture;
+        this.corporateCardSurcharge = builder.corporateCardSurcharge;
+        this.totalAmount = builder.totalAmount;
+        this.refundSummary = builder.refundSummary;
+        this.settlementSummary = builder.settlementSummary;
+        this.cardDetails = builder.cardDetails;
+        this.links = builder.links;
+        this.metadata = builder.metadata == null || builder.metadata.isEmpty() ? Optional.empty() : Optional.of(builder.metadata);;
```

Comment:
Just realised that the Optional wrapping can move to the getter now if you'd prefer to keep the constructor simpler.